app/src/test/java/com/android/sample/model/network/ActivityDaoTest.kt

```diff
@@ -0,0 +1,65 @@
+package com.android.sample.model.network
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.room.Room
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import com.android.sample.model.activity.ActivityStatus
+import com.android.sample.model.activity.database.ActivityDao
+import com.android.sample.model.activity.database.AppDatabase
+import com.android.sample.resources.dummydata.activity1
+import com.android.sample.resources.dummydata.activity2
+import kotlinx.coroutines.runBlocking
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class ActivityDaoTest {
+
+  @get:Rule val instantTaskExecutorRule = InstantTaskExecutorRule()
+
+  private lateinit var database: AppDatabase
+  private lateinit var activityDao: ActivityDao
+
+  @Before
+  fun setUp() {
+    val context = InstrumentationRegistry.getInstrumentation().targetContext
+    database =
+        Room.inMemoryDatabaseBuilder(context, AppDatabase::class.java)
+            .allowMainThreadQueries()
+            .build()
+    activityDao = database.activityDao()
+  }
+
+  @After
+  fun tearDown() {
+    database.close()
+  }
+
+  @Test
+  fun insertAndRetrieveActivities() = runBlocking {
+    val activity = activity1
+    activityDao.insertActivities(listOf(activity))
+
+    val retrievedActivities = activityDao.getActivitiesByIds(listOf(activity.uid))
+    assertEquals(listOf(activity), retrievedActivities)
+  }
+
+  @Test
+  fun clearActivitiesByStatus() = runBlocking {
+    val activity1 = activity1
```

Comment:
There is no need to instantiate the activities already present in the DummyData file, just call them